autoscale/src/main/java/com/sequenceiq/periscope/repository/ClusterRepository.java

```diff
@@ -25,6 +29,9 @@
     @Query("SELECT c FROM Cluster c LEFT JOIN FETCH c.clusterPertain WHERE c.clusterPertain.userId = :userId")
     List<Cluster> findByUserId(@Param("userId") String userId);
 
+    @Query("SELECT c FROM Cluster c LEFT JOIN FETCH c.clusterPertain WHERE c.clusterPertain.userId = :userId and c.stackType = :stackType")
```

Comment:
Same as previous - significance of the 'stackType' field? Why was it not required earlier when the id was being used?

Reply:
It was not required in earlier CB, but since now we differentiate DataHub clusters and Datalake clusters this is required whether id based or crn based.

Reply:
Sounds good.